### PR TITLE
MM-15864 - Made Post.GetRepliesForExport sync

### LINKED ISSUE #10964: Migrate "Post.GetRepliesForExport" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15864)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetRepliesForExport` in the `Post` store.

The expected way to implement it is, go to the `Post` store implementation in the `store/sqlstore/` directory, modify the method `GetRepliesForExport` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by willdot**
#### Summary
This PR made changes to Post.GetRepliesForExport to be sync and made changes to code that called this function, to no longer call it async as a channel.

(This is my first contribution to OSS, so sorry if more detail is needed. Let me know if it is)

Fixes #10964

https://mattermost.atlassian.net/browse/MM-15864

### CODE REVIEW

**Comment by hanzei** [comment]
@willdot Would you be open on signing the [contribution license agreement](https://www.mattermost.org/mattermost-contributor-agreement/)?

**Comment by willdot** [comment]
I signed the agreement the other day, but looking at the list of usernames, auto correct put a space my username. Have signed it again without the space :)

**Comment by hanzei** [comment]
Thanks @willdot :+1:

File: app/export.go [hunk 1]
```diff
@@ -379,15 +379,13 @@ func (a *App) ExportAllPosts(writer io.Writer) *model.AppError {
 func (a *App) buildPostReplies(postId string) (*[]ReplyImportData, *model.AppError) {
 	var replies []ReplyImportData
 
-	result := <-a.Srv.Store.Post().GetRepliesForExport(postId)
+	result, err := a.Srv.Store.Post().GetRepliesForExport(postId)
```
Comment (lieut-data) [inline app/export.go]: nit: can we rename `result` to the more useful `replyPosts`, now that we don't have a generic result container?

Comment (willdot) [reply app/export.go]: Of course 👍

Comment (willdot) [reply app/export.go]: @lieut-data @hanzei 

Hey, I can't figure out why the commit couldn't be built?

Comment (lieut-data) [reply app/export.go]: Looks like it worked eventually!

**Pull request merged**